service_provider.go

```diff
@@ -1656,18 +1656,24 @@ func findChild(parentEl *etree.Element, childNS string, childTag string) (*etree
 }
 
 func elementToBytes(el *etree.Element) ([]byte, error) {
+	// Retrieve namespaces from the element itself and its parents
 	namespaces := map[string]string{}
-	for _, childEl := range el.FindElements("//*") {
-		ns := childEl.NamespaceURI()
-		if ns != "" {
-			namespaces[childEl.Space] = ns
+	currentElement := el
+	for currentElement != nil {
```

Comment:
Could you please add a new test case for testing this functionality (elements from the default namespace)?

Reply:
Added a test for it 👍